rfcs/62-editor-data-model.md

```diff
@@ -0,0 +1,653 @@
+
+<!-- Explain the proposal as if it was already included in the engine and you were teaching it to another Bevy user. That generally means:
+
+- Introducing new named concepts.
+- Explaining the feature, ideally through simple examples of solutions to concrete problems.
+- Explaining how Bevy users should *think* about the feature, and how it should impact the way they use Bevy. It should explain the impact as concretely as possible.
+- If applicable, provide sample error messages, deprecation warnings, or migration guidance.
+- If applicable, explain how this feature compares to similar existing features, and in what situations the user would use each one. -->
+
+## Implementation strategy
+
+<!-- This is the technical portion of the RFC.
+Try to capture the broad implementation strategy,
+and then focus in on the tricky details so that:
+
+- Its interaction with other features is clear.
+- It is reasonably clear how the feature would be implemented.
+- Corner cases are dissected by example.
+
+When necessary, this section should return to the examples given in the previous section and explain the implementation details that make them work.
+
+When writing this section be mindful of the following [repo guidelines](https://github.com/bevyengine/rfcs):
+
+- **RFCs should be scoped:** Try to avoid creating RFCs for huge design spaces that span many features. Try to pick a specific feature slice and describe it in as much detail as possible. Feel free to create multiple RFCs if you need multiple features.
+- **RFCs should avoid ambiguity:** Two developers implementing the same RFC should come up with nearly identical implementations.
+- **RFCs should be "implementable":** Merged RFCs should only depend on features from other merged RFCs and existing Bevy features. It is ok to create multiple dependent RFCs, but they should either be merged at the same time or have a clear merge order that ensures the "implementable" rule is respected. -->
+
+### Names
+
+Being data-heavy, and forfeiting the structural access of Rust types, the data model makes extensive use of strings to reference items. In order to make those operations efficient, we define a `Name` type to represent those strings. The implementation of `Name` is outside of the scope of this RFC and left as an optimization, and should be considered as equivalent to a `String` wrapper:
+
+```rust
+struct Name(pub String);
+// -or-
+type Name = String;
+```
+
+The intent is to explore some optimization strategies like string interning for fast comparison. Those optimizations are left out of the scope of this RFC.
+
+### Built-in types
+
+The data model supports a number of built-in types, corresponding to common Rust built-in types.

```

Comment:
Base types? I really don't know how else to call them, they're literally the types built in the data model, that allow creating other types.

Reply:
Hmm, I think we can still call them built-in types, but we need to call Rust primitive types with their actual name to avoid confusion.

```suggestion
The data model supports a number of built-in types, corresponding to common Rust primitive types.
```